x-pack/test/functional/services/cases/list.ts

```diff
@@ -85,10 +84,13 @@ export function CasesTableServiceProvider(
     },
 
     async validateCasesTableHasNthRows(nrRows: number) {
-      await retry.tryForTime(3000, async () => {
+      await retry.waitFor(`the cases table to have ${nrRows} cases`, async () => {
+        await this.refreshTable();
```

Comment:
Sorry, I approved but just thought about this now.

Is it possible that this refresh takes longer than the timeout of the `find` in the next line?

Reply:
Good question! I do not think so but I am not 100% sure. Do you think we should approach it differently?

Reply:
Hard to say, maybe moving the `refreshTable` outside the `waitFor`. I wonder if as things are currently every retry will start with a `refreshTable`.

Reply:
We use the same pattern below in `waitForCasesToBeListed` this is why I added it. It is tested through time (all flaky tests are green). I noticed that the `refreshTable` just calls a button but does not wait for anything. Wdyt should we remove it?

Reply:
> Btw I noticed that the `refreshTable` just calls a button but does not wait for anything.

Yeah, I checked it too that was why I mentioned it 😅  Flaky tests are really annoying so that's why I was going over what might cause issues.

If tests are passing I am ok with leaving it as is for now. If the issue ever comes back we have something to look at 🤷

Reply:
You concerns are valid. I removed it in https://github.com/elastic/kibana/pull/170383/commits/6bece25adf169169358581d2249c176dcdfa9390.